examples/comparison_gfunction_solvers.py

```diff
@@ -150,52 +156,52 @@ def main():
-    ax.set_ylabel(r'Absolute error')
-    gt.utilities._format_axes(ax)
-    # Absolute error
-    ax.plot(lntts, np.abs(gfunc_equivalent.gFunc - gfunc_similarities.gFunc),
-            label='equivalent')
-    ax.legend()
-    ax.set_title(f"Absolute error relative to the 'similarities' solver "
-                 f"(Field of {N_1} by {N_2} boreholes)")
-    # Adjust to plot window
-    fig.tight_layout()
-
-    # Draw relative error
-    # Configure figure and axes
-    fig = gt.utilities._initialize_figure()
-    ax = fig.add_subplot(111)
-    # Axis labels
-    ax.set_xlabel(r'ln$(t/t_s)$')
-    ax.set_ylabel(r'Relative error')
-    gt.utilities._format_axes(ax)
-    # Relative error
-    ax.plot(lntts, (gfunc_equivalent.gFunc - gfunc_similarities.gFunc) / gfunc_similarities.gFunc,
-            label='equivalent')
-    ax.legend()
-    ax.set_title(f"Relative error relative to the 'similarities' solver "
-                 f"(Field of {N_1} by {N_2} boreholes)")
-    # Adjust to plot window
-    fig.tight_layout()
-
-    return
+    if make_plots:
+
+        # -------------------------------------------------------------------------
+        # Plot results
+        # -------------------------------------------------------------------------
+        # Draw g-functions
+        ax = gfunc_similarities.visualize_g_function().axes[0]
+        ax.plot(lntts, gfunc_equivalent.gFunc, 'ro')
+        ax.legend([f'similarities (t = {t_similarities:.3f} sec)',
+                f'equivalent (t = {t_equivalent:.3f} sec)'])
+        ax.set_title(f'Field of {N_1} by {N_2} boreholes')
+        plt.tight_layout()
+
+        # Draw absolute error
+        # Configure figure and axes
+        fig = gt.utilities._initialize_figure()
+        ax = fig.add_subplot(111)
+        # Axis labels
+        ax.set_xlabel(r'ln$(t/t_s)$')
+        ax.set_ylabel(r'Absolute error')
+        gt.utilities._format_axes(ax)
+        # Absolute error
+        ax.plot(lntts, np.abs(gfunc_equivalent.gFunc - gfunc_similarities.gFunc),
+                label='equivalent')
+        ax.legend()
+        ax.set_title(f"Absolute error relative to the 'similarities' solver "
+                    f"(Field of {N_1} by {N_2} boreholes)")
+        # Adjust to plot window
+        fig.tight_layout()
+
+        # Draw relative error
+        # Configure figure and axes
+        fig = gt.utilities._initialize_figure()
+        ax = fig.add_subplot(111)
+        # Axis labels
+        ax.set_xlabel(r'ln$(t/t_s)$')
+        ax.set_ylabel(r'Relative error')
+        gt.utilities._format_axes(ax)
+        # Relative error
+        ax.plot(lntts, (gfunc_equivalent.gFunc - gfunc_similarities.gFunc) / gfunc_similarities.gFunc,
+                label='equivalent')
+        ax.legend()
+        ax.set_title(f"Relative error relative to the 'similarities' solver "
+                    f"(Field of {N_1} by {N_2} boreholes)")
+        # Adjust to plot window
+        fig.tight_layout()
```

Comment:
Is there any chance to compile all the plotting into one place?  Maybe a single file/class called plotting.py?  There seems to be quite a bit of repetition, but each of these also seems a little different.  So no biggie if not.  Just a thought.